File: backend/Models/Entities/Attendance.cs
namespace backend.Models.Entities
{
    public class Attendance
    {
        public int Id { get; set; }

        public string EmployeeId { get; set; } = null!;

        public DateTime CheckInTime { get; set; }

        public DateTime? CheckOutTime { get; set; }

        public DateTime WorkDate { get; set; }

        public string? Status { get; set; }

        public string? DeviceType { get; set; }

        public Employee Employee { get; set; } = null!;
    }
}

File: frontend/src/api/models/attendance.ts
export interface UpdateAttendanceRequest {
  checkInTime: string;
  checkOutTime: string;
  workDate: string;
}

export interface Attendance {
  id: number;
  employeeId: string;
  employeeCode: string;
  employeeName: string;
  checkInTime: string;
  checkOutTime: string;
  workDate: string;
  status: string;
  deviceType: string;
}

export interface AttendanceQueryParams {
  PageNumber?: number;
  PageSize?: number;
  employeeId?: string;
  startDate?: string;
  endDate?: string;
}